mule-extensions-api/src/main/java/org/mule/extension/api/introspection/property/ImportedTypesModelProperty.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.introspection.property;
+
+
+import org.mule.extension.api.introspection.ExtensionModel;
+import org.mule.extension.api.introspection.ModelProperty;
+import org.mule.metadata.api.model.MetadataType;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * An immutable model property which specifies the {@link MetadataType} used by
+ * the {@link ExtensionModel} that have to be imported from a different {@link ExtensionModel}
+ *
+ * @since 1.0
+ */
+public final class ImportedTypesModelProperty implements ModelProperty
+{
+
+    private final Map<MetadataType, Class<?>> importedTypes;
```

Comment:
value should also be a MetadataType. No externalizable ModelProperty should ever referenced a java or javax. package